src/customKeywords/files.js

```diff
@@ -0,0 +1,46 @@
+var Ajv = require('ajv');
+const fileNameField = 'fieldname';
+
+function missingElements(superSet, subset) {
+    let missingElements = [];
+    subset.every(function (subsetElement) {
+        if (!superSet.includes(subsetElement)) {
+            missingElements.push(subsetElement);
+        }
+        return true;
+    });
+
+    return missingElements;
+};
+
+module.exports = {
+    validate: function filesValidation(schema, data) {
+        filesValidation.errors = [];
+        const dataFileName = data.map((element) => { return element[fileNameField] });
+        let missingFiles = missingElements(dataFileName, schema.required);
+        if (missingFiles.length > 0) {
+            filesValidation.errors.push(new Ajv.ValidationError({
+                keyword: 'files',
+                message: 'Missing required files: ' + schema.required.toString(),
+                params: { requiredFiles: schema.required, missingFiles: missingFiles }
+            }));
+            return false;
+        }
+
+        // Check that only the optional files exists
+        let allFiles = schema.required.concat(schema.optional);
+        let extraFiles = missingElements(allFiles, dataFileName);
+        if (extraFiles.length > 0) {
```

Comment:
shouldn't this check be governed by "additionalProperties" flag?

Reply:
do you mean if the spec allows additional not defined files?

Can you please provide a valid open API spec snippet for that?

Reply:
Looked it up, actually I think that I am wrong - additionalProperties can be defined for schema, but not for endpoint, so please ignore my comment.